domain/application/errors/errors.go

```diff
@@ -174,4 +174,9 @@ const (
 	// CharmProvenanceNotValid describes an error that occurs when the
 	// charm download provenance is not valid.
 	CharmProvenanceNotValid = errors.ConstError("charm provenance not valid")
+
+	// InvalidResourceArgs indicates the provided resource arguments are not
+	// valid, for instance when we try to override non-existing resources  in
+	// the charm.
+	InvalidResourceArgs = errors.ConstError("invalid resource args")
```

Comment:
It is an interesting opinion. I followed the trend, but yeah, I think you have a point.

Reply:
Please don't use `gc.ErrorMatches`, If you care about the error, in testing or elsewhere, return the error.

Reply:
At least if you throw this error, you can easily see that most likely it's a programming error. Otherwise it's easy to ignore a pass back to a user. It should never reach the user.